src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (!target || !(target instanceof Herbivore)) {
```

Comment:
The condition here checks if the target is not an instance of Herbivore, but it doesn't check if it's an instance of Carnivore. The method should not work if the target is another Carnivore, so you should add a check to ensure that the target is not an instance of Carnivore.